### PR TITLE
fix(pattern): add check for absurd patterns

### LINKED ISSUE #775: We didn't handle absurd clauses without absurd patterns
```
data False
open data Truth | rua
def f Truth : False
| rua
def inconsistency : False => f rua
```

### PR DESCRIPTION
**Pull request by SchrodingerZhu**
- Gate away the reachable match arms with empty body from passing type-checking:
- Check: for those patterns used in elimination (rather than data decl), if Pattern.Absurd () does not appear on lhs, then we require a body

close #775

### CODE REVIEW

**Review by ice1000 — APPROVED** [review]
bors r+

**Review by ice1000 — COMMENTED** [review]
bors r-

File: base/src/main/java/org/aya/tyck/pat/PatternProblem.java [hunk 1]
```diff
@@ -143,4 +143,20 @@ record BadLitPattern(
         Doc.par(1, type.toDoc(options)));
     }
   }
+
+  record InvalidEmptyBody(
+    @NotNull Pattern.Clause match
+  ) implements PatternProblem {
+    @Override @NotNull public Pattern pattern() {
+      return match.patterns.first().term();
+    }
```
Comment (ice1000) [inline base/src/main/java/org/aya/tyck/pat/PatternProblem.java:152]:
```suggestion
  ) implements Problem {
```

**Review by ice1000 — CHANGES_REQUESTED** [review]
bors r-

File: base/src/main/java/org/aya/tyck/pat/PatternProblem.java (hunk 1 above)
Comment (SchrodingerZhu) [reply base/src/main/java/org/aya/tyck/pat/PatternProblem.java]: resolved

**Review by ice1000 — APPROVED** [review]
bors r+

**Pull request merged**